ground/src/test/java/com/google/android/ground/ui/home/HomeScreenViewModelTest.kt

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright 2024 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.ground.ui.home
+
+import com.google.android.ground.BaseHiltTest
+import com.google.android.ground.persistence.local.stores.LocalOfflineAreaStore
+import com.google.android.ground.testNavigateTo
+import com.google.android.ground.ui.common.Navigator
+import com.sharedtest.FakeData.OFFLINE_AREA
+import dagger.hilt.android.testing.HiltAndroidTest
+import javax.inject.Inject
+import kotlinx.coroutines.test.advanceUntilIdle
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.robolectric.RobolectricTestRunner
+
+@HiltAndroidTest
+@RunWith(RobolectricTestRunner::class)
+class HomeScreenViewModelTest : BaseHiltTest() {
+
+  @Inject lateinit var navigator: Navigator
+  @Inject lateinit var homeScreenViewModel: HomeScreenViewModel
+  @Inject lateinit var localOfflineAreaStore: LocalOfflineAreaStore
+
+  @Test
+  fun `showAbout is navigating to correct screen`() = runWithTestDispatcher {
+    testNavigateTo(navigator.getNavigateRequests(), HomeScreenFragmentDirections.showAbout()) {
+      homeScreenViewModel.showAbout()
+    }
```

Comment:
Can we perform UI click instead of calling the nav directions?

Reply:
ViewModel Testing vs Fragment Testing

Reply:
@sufyanAbbasi What is your opinion on testing fragments vs view models? I feel testing via fragment gives a better coverage including the UI components. If we add a test for view model right now, that part would still remain missing and having both fragment and view model unit tests would result in duplicate tests.

Reply:
Putting my thoughts here:

1. In terms of coverage, yes the above lines are correct
2. Regarding testing strategy, we have two things, for example, A. writing test cases to check Is my view visible on screen or not, B. Are my ViewModel functions working as expected or not, are my ViewModel variables are having correct values or not, no matter how they are consumed by UI or not, but values in variables are correct or not. 
That's why, it's important to have Both the test cases to make sure all pieces of code are working correctly independently.

Reply:
discussed offline, we'll be going ahead with adding both coverages if possible.